core/src/main/java/org/apache/druid/data/input/MapBasedRow.java

```diff
@@ -43,7 +45,7 @@ public MapBasedRow(
   )
   {
     this.timestamp = timestamp;
-    this.event = event;
+    this.event = Collections.unmodifiableMap(Preconditions.checkNotNull(event, "event"));
```

Comment:
Nit - Is it better to use Guava's ImmutableMap here?  It seems to offer [stronger immutability](https://stackoverflow.com/a/22636750/5423491) (though this might not matter since the `event` argument is unlikely to be modified in this constructor) and seems to be more widely used across the Druid codebase.

Reply:
`ImmutableMap` only offers `copyOf()` method to create it from an existing map which copies all elements in the original map to the immutableMap. This can impact to the query performance.